pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDaoTest.scala

```diff
@@ -35,56 +37,124 @@ class IdentifiersDaoTest
       testWith((identifiersDao, identifiersTable))
     }
 
-  describe("lookupID") {
-    it("gets an Identifier if it finds a matching SourceSystem and SourceId") {
+  describe("lookupIds") {
+
+    it("retrieves a single id from the identifiers database") {
       val sourceIdentifier = createSourceIdentifier
       val identifier = createSQLIdentifierWith(
         sourceIdentifier = sourceIdentifier
       )
 
       withIdentifiersDao {
         case (identifiersDao, _) =>
-          identifiersDao.saveIdentifier(identifier) shouldBe Success(1)
+          val triedLookup = for {
+            _ <- identifiersDao.saveIdentifiers(List(identifier))
+            lookupResult <- identifiersDao.lookupIds(List(sourceIdentifier))
+          } yield (lookupResult)
+
+          triedLookup shouldBe a[Success[_]]
+          triedLookup.get.found shouldBe Map(sourceIdentifier -> identifier)
+          triedLookup.get.notFound shouldBe empty
+      }
+    }
 
-          val triedLookup = identifiersDao.lookupId(
-            sourceIdentifier = sourceIdentifier
-          )
+    it("retrieves multiple ids from the identifiers database") {
+      val identifiersMap = (1 to 3).map { _ =>
+        val sourceIdentifier = createSourceIdentifier
+        val identifier =
+          createSQLIdentifierWith(sourceIdentifier = sourceIdentifier)
+        (sourceIdentifier, identifier)
+      }.toMap
 
-          triedLookup shouldBe Success(Some(identifier))
+      withIdentifiersDao {
+        case (identifiersDao, _) =>
+          val triedLookup = for {
+            _ <- identifiersDao.saveIdentifiers(identifiersMap.values.toList)
+            lookupResult <- identifiersDao.lookupIds(identifiersMap.keys.toList)
+          } yield (lookupResult)
+
+          triedLookup shouldBe a[Success[_]]
+          triedLookup.get.found shouldBe identifiersMap
+          triedLookup.get.notFound shouldBe empty
       }
     }
 
     it(
-      "does not get an identifier if there is no matching SourceSystem and SourceId") {
-      val identifier = createSQLIdentifier
+      "matches correctly the ids to the sourceIdentifier when the sourceIdentifiers have overlapping values") {
+      val sourceIdentifier = createSourceIdentifier
+      val overlappingSourceIdentifier =
+        sourceIdentifier.copy(ontologyType = "Item")
```

Comment:
This works as long as `createSourceIdentifier` picks an ontology that isn't `Item`.

Make the difference more explicit, and get names that line up nicely to boot:

```suggestion
      val workSourceIdentifier = createSourceIdentifierWith(ontologyType = "Work")
      val itemSourceIdentifier = workSourceIdentifier(ontologyType = "Item")
```